rpcs3/rpcs3qt/render_creator.cpp

```diff
@@ -88,7 +87,8 @@ render_creator::render_creator(QObject *parent) : QObject(parent)
 	{
 		supports_vulkan = device_found;
 		vulkan_adapters = std::move(compatible_gpus);
-		enum_thread.join();
+		(*enum_thread)();
```

Comment:
Looks really bad. I'm sure there's a better syntax

Reply:
Btw @elad335 since we don't detach named_threads, what happens when this function exits while the check is stuck? Aren't we just leaking it? Feels like it should get stuffed into fxo or something for cleanup later.

Reply:
At least with std::thread the object would clean up after itself on exit, but I'm pretty sure named_thread doesn't (and for good reason).

Reply:
named_thread destructor always joins the thread, cleanup is not allowed for it in current master.

Reply:
Since this is qt class, maybe just try QThread?
Idk if it can handle the problem in a better way

Reply:
Probably too much hassle

Reply:
Currently thread_ctrl::emergancy_exit is not terminating RPCS3 only for named_thread.

Reply:
> named_thread destructor always joins the thread, cleanup is not allowed for it in current master.

This statement just makes the code here stranger - if dtor will join the thread, then why invoke operator() at all? I mean, we have a delete immediately after which will call the dtor anyway.

Reply:
Also at least add a TODO or something about this being an intentional leak. If the func never completes, dtor will never be called for it and the object stays unaccounted for which makes me squirm.

Reply:
dtor also signals "abort execution" to the thread before joining it but in this case the thread already completed its job so it should be OK.

Reply:
Used std::unique_ptr and itse release() function for it to be much more explicit.

Reply:
Thanks. That makes it more obvious what's going on.